module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/RecordsHelperTest.php

```diff
@@ -0,0 +1,339 @@
+        $loader = $this->createMock(Loader::class);
+
+        $records = [
+            ['source' => 'Solr', 'title' => 'Test Book'],
+        ];
+
+        $expectedIds = [
+            ['id' => '', 'source' => 'Solr'],
+        ];
+
+        $driver = $this->createMock(AbstractBase::class);
+        $mockDrivers = [$driver];
+
+        $loader->expects($this->once())
+            ->method('loadBatch')
+            ->with($expectedIds, true)
+            ->willReturn($mockDrivers);
+
+        $driver->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[0]);
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->getDrivers($records);
+
+        $this->assertIsArray($result);
+        $this->assertCount(1, $result);
+    }
+
+    /**
+     * Test getDrivers with records using default search backend
+     *
+     * @return void
+     */
+    public function testGetDriversWithDefaultSearchBackend(): void
+    {
+        $config = $this->createMock(Config::class);
+        $loader = $this->createMock(Loader::class);
+
+        $records = [
+            ['id' => 'record1', 'title' => 'Test Book'],
+        ];
+
+        $expectedIds = [
+            ['id' => 'record1', 'source' => DEFAULT_SEARCH_BACKEND],
+        ];
+
+        $driver = $this->createMock(AbstractBase::class);
+        $mockDrivers = [$driver];
+
+        $loader->expects($this->once())
+            ->method('loadBatch')
+            ->with($expectedIds, true)
+            ->willReturn($mockDrivers);
+
+        $driver->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[0]);
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->getDrivers($records);
+
+        $this->assertIsArray($result);
+        $this->assertCount(1, $result);
+    }
+
+    /**
+     * Test collectRequestStats when ajax is enabled
+     *
+     * @return void
+     */
+    public function testCollectRequestStatsWithAjaxEnabled(): void
+    {
+        $config = new Config(['Authentication' => ['enableAjax' => true]]);
+        $loader = $this->createMock(Loader::class);
+
+        $ilsDetails1 = ['id' => 'record1', 'status' => 'available'];
+        $ilsDetails2 = ['id' => 'record2', 'status' => 'pending'];
+
+        $driver1 = $this->createMock(AbstractBase::class);
+        $driver2 = $this->createMock(AbstractBase::class);
+
+        $driver1->expects($this->once())
+            ->method('getExtraDetail')
+            ->with('ils_details')
+            ->willReturn($ilsDetails1);
+
+        $driver2->expects($this->once())
+            ->method('getExtraDetail')
+            ->with('ils_details')
+            ->willReturn($ilsDetails2);
+
+        $records = [$driver1, $driver2];
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats($records);
+
+        $this->assertIsArray($result);
```

Comment:
Can we make a more specific assertion here, like:


```suggestion
        $this->assertSame([$ilsDetails1, $ilsDetails], $result);
```

...? Or is there a reason we can't?